[SuiteSparse] Update and split into sub-packages (v2)

valgur opened the pull request:
Continues #39297.

Additional changes in this PR:
- Moved update_suitesparse.py to scripts/.
- Made the "gpl" feature for `suitesparse` disabled by default.
- Dropped the "complex" feature from `suitesparse-cxsparse`. This leaves it enabled by default, where it's supported - everything except MSVC.
- Renamed the "compact" (default-enabled) feature to "precompiled" (default-disabled) in `suitesparse-graphblas` to make it easier to use.
- Updated to the latest release (v7.8.3).

Review comments that did not get addressed:
- The default JIT support in suitesparse-graphblas is likely still problematic, especially for cross-compilation.
- I did not move the changes related to the replacement of `suitesparse` dependency with its individual components in Ceres and elsewhere to a separate PR.

Let me know if I missed anything.

Fixes #24252.

----

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] SHA512s are updated for each updated download.
- [x] The "supports" clause reflects platforms that may be fixed by this new version.
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

valgur committed:
[SuiteSparse] add suitesparse-config port

valgur committed:
[SuiteSparse] add suitesparse-amd

valgur committed:
[SuiteSparse] add suitesparse-btf

valgur committed:
[SuiteSparse] add suitesparse-camd

valgur committed:
[SuiteSparse] add suitesparse-ccolamd

valgur committed:
[SuiteSparse] add suitesparse-colamd

valgur committed:
[SuiteSparse] add suitesparse-ldl

valgur committed:
[SuiteSparse] add suitesparse-mongoose

valgur committed:
[SuiteSparse] add suitesparse-rbio

valgur committed:
[SuiteSparse] add suitesparse-spex

valgur committed:
[SuiteSparse] add suitesparse-graphblas

valgur committed:
[SuiteSparse] add suitesparse-lagraph

valgur committed:
[SuiteSparse] add suitesparse-cholmod

valgur committed:
[SuiteSparse] add suitesparse-klu

valgur committed:
[SuiteSparse] add suitesparse-umfpack

valgur committed:
[SuiteSparse] add suitesparse-paru

valgur committed:
[SuiteSparse] add suitesparse-spqr

valgur committed:
[SuiteSparse] add suitesparse-cxsparse

valgur committed:
[SuiteSparse] fix suitesparse-ldl license installation

valgur committed:
[SuiteSparse] update suitesparse meta-package

valgur committed:
[SuiteSparse] update Ceres and g2o to use specific SuiteSparse packages only

valgur committed:
[SuiteSparse] fix vcpkg.json formatting

valgur committed:
[SuiteSparse] update version database

valgur committed:
[SuiteSparse] fix suitesparse-config fixup

valgur committed:
[SuiteSparse] vdb

valgur committed:
[SuiteSparse] set DISABLE_PARALLEL_CONFIGURE

Parallel configure fails due to race conditions with configure_file().

valgur committed:
[SuiteSparse] vdb

valgur committed:
[SuiteSparse] disable OpenMP by default

valgur committed:
[SuiteSparse] update vdb

valgur committed:
[SuiteSparse] fix port-version

valgur committed:
[SuiteSparse] update openmp in suitesparse/vcpkg.json

valgur committed:
[SuiteSparse] workaround for find_package(LAPACK) issues due to BLA_VENDOR

valgur committed:
[Ceres] add compatibility with SuiteSparse v7

Commit:
update git-tree

valgur committed:
[Ceres] update vdb

valgur committed:
[Ceres] add missing suitesparse-cxsparse dependency

valgur committed:
[Ceres] update vdb

valgur committed:
[Ceres] fix CeresConfig.cmake.in

valgur committed:
[Ceres] update vdb

valgur committed:
[SuiteSparse] add missing openmp feature to lagraph

valgur committed:
[SuiteSparse] update vdb

valgur committed:
[theia] add support for SuiteSparse v7

valgur committed:
[theia] update vdb

valgur committed:
[Theia] update and merge patches, drop default [viewer] from openimageio

valgur committed:
[Theia] update vdb

valgur committed:
[SuiteSparse] handle suitesparse_mongoose tool

valgur committed:
[SuiteSparse] update vdb

valgur committed:
[SuiteSparse] fix GraphBLAS JIT cache dir handling

valgur committed:
[SuiteSparse] update vdb

valgur committed:
[SuiteSparse] disable complex data types for ARM android

valgur committed:
vdb

valgur committed:
[suitesparse] fix BLAS linking on Windows

valgur committed:
[suitesparse-cxsparse] complex is not available on MSVC

valgur committed:
[suitesparse] fix BLAS linking in components using it transitively

valgur committed:
vdb

valgur committed:
[suitesparse-graphblas] disable on android

GrpahBLAS has a strict requirement for complex math support, which Android NDK seems to lack.

valgur committed:
vdb

valgur committed:
[suitesparse-paru] MSVC OpenMP is not compatible (v2.0 < required v4.5)

valgur committed:
[suitesparse-cxsparse] strictly disable 'complex' on Windows

valgur committed:
[suitesparse] disable CUDA feature on Windows

Don't have a good enough setup to debug this configuration currently.
Works locally on Linux, though.

valgur committed:
[suitesparse-graphblas] disable x86 Windows

unresolved external symbol __InterlockedOr64

valgur committed:
[suitesparse] fix vcpkg.json formatting

valgur committed:
vdb

valgur committed:
[suitesparse-cxsparse] don't enable 'complex' on Windows

valgur committed:
[suitesparse] fix 'cuda' feature support

'matrixops' feature is required in CHOLMOD and SPQR was not packaging two CMake config files correctly

valgur committed:
vdb

valgur committed:
[colmap] add a missing METIS dependency

Used to be provided transitively by SuiteSparse before.

valgur committed:
vdb

valgur committed:
[suitesparse-graphblas] add x86-windows to ci.baseline.txt

valgur committed:
Bump port-versions

valgur committed:
update vdb

valgur committed:
Merge remote-tracking branch 'refs/remotes/origin/master' into update/suitesparse

valgur committed:
CI

valgur committed:
Merge remote-tracking branch 'origin/master' into update/suitesparse

# Conflicts:
#	ports/colmap/vcpkg.json
#	versions/baseline.json
#	versions/c-/colmap.json

valgur committed:
vdb

valgur committed:
[suitesparse-graphblas] only supported on x64 for windows

valgur committed:
vdb

valgur committed:
[SuiteSparse] update to v7.8.1

valgur committed:
vdb

valgur committed:
[suitesparse-spqr] add a missing 'supernodal' requirement for cholmod

valgur committed:
vdb

valgur committed:
[ceres] require matrixops feature from cholmod

valgur committed:
vdb

valgur committed:
[suitesparse-graphblas] move failures from "supports" to ci.baseline.txt

valgur committed:
vdb

valgur committed:
fix ceres vdb

valgur committed:
fix g2o vdb

valgur committed:
[suitesparse-graphblas] adjust "supports"

valgur committed:
vdb

valgur committed:
[g2o] disable use of vendored CSparse

valgur committed:
vdb

Commit:
Merge branch 'master' of https://github.com/microsoft/vcpkg into update/suitesparse

valgur committed:
[SuiteSparse] add update_suitesparse.py script

valgur committed:
[SuiteSparse] update to v7.8.2

valgur committed:
[SuiteSparse] use lower-case `share/suitesparse` subdir

valgur committed:
vdb

valgur committed:
[theia] drop REQUIRED from `find_dependency(Ceres REQUIRED)`

valgur committed:
vdb

valgur committed:
Merge remote-tracking branch 'origin/master' into update/suitesparse

# Conflicts:
#	versions/t-/theia.json

valgur committed:
update vdb

valgur committed:
[SuiteSparse] move update_suitesparse.py to scripts/

valgur committed:
[SuiteSparse] update to v7.8.3

valgur committed:
update vdb

valgur committed:
[suitesparse] don't enable "gpl" feature by default

valgur committed:
[suitesparse-cxsparse] drop the "complex" feature

Allow the project to auto-detect it based on whether it's being compiled with MSVC or not.

valgur committed:
[suitesparse-graphblas] rename "compact" -> "precompiled"

valgur committed:
vdb

valgur commented:
cc @vicroms @MonicaLiu0311 @dg0yt

dg0yt commented on ports/ceres/0003_fix_exported_ceres_config.patch:
```diff
@@ -2,7 +2,24 @@ diff --git a/cmake/CeresConfig.cmake.in b/cmake/CeresConfig.cmake.in
+-@CXSparse_DEPENDENCY@
+-@SuiteSparse_DEPENDENCY@
++include (CMakeFindDependencyMacro)
++if (@CXSPARSE@)
++    find_dependency(CXSparse REQUIRED)
++endif()
++if (@SUITESPARSE@)
++    find_dependency(CHOLMOD REQUIRED)
++    find_dependency(SPQR REQUIRED)
++endif()
```
Do not use `REQUIRED` with `find_dependency()`.
```suggestion
+if (@CXSPARSE@)
+    find_dependency(CXSparse)
+endif()
+if (@SUITESPARSE@)
+    find_dependency(CHOLMOD)
+    find_dependency(SPQR)
+endif()
```

dg0yt commented on ports/suitesparse-cholmod/portfile.cmake:
```diff
@@ -0,0 +1,68 @@
+)
+
+vcpkg_check_features(OUT_FEATURE_OPTIONS FEATURE_OPTIONS
+    FEATURES
+    matrixops  CHOLMOD_MATRIXOPS
+    modify     CHOLMOD_MODIFY
+    partition  CHOLMOD_PARTITION
+    supernodal CHOLMOD_SUPERNODAL
+    openmp     CHOLMOD_USE_OPENMP
```
```suggestion
    FEATURES
        matrixops  CHOLMOD_MATRIXOPS
        modify     CHOLMOD_MODIFY
        partition  CHOLMOD_PARTITION
        supernodal CHOLMOD_SUPERNODAL
        openmp     CHOLMOD_USE_OPENMP
```

dg0yt commented on ports/suitesparse-cholmod/portfile.cmake:
```diff
@@ -0,0 +1,68 @@
+    openmp     CHOLMOD_USE_OPENMP
+)
+
+set(CUDA_ENABLED OFF)
+if("cuda" IN_LIST FEATURES)
+    set(CUDA_ENABLED ON)
+    set(CUDA_ARCHITECTURES "native")
+endif()
+if("cuda-redist" IN_LIST FEATURES)
+    set(CUDA_ENABLED ON)
+    set(CUDA_ARCHITECTURES "all-major")
+endif()
```
IMO there should be only a single `cuda`, probably with `all-major` architectures.
(Similar to `-march=native`, `native`  architectures would be a triplet customization.)

valgur replied:
Ah, good point regarding the triplet file. I agree that it's a more appropriate place for customizations.

I based this off the colmap port: https://github.com/microsoft/vcpkg/blob/4f746bc66438fce2b900c3ba6094a483b871b045/ports/colmap/vcpkg.json#L39-L51
Perhaps that one should be adjusted accordingly in the future as well.

dg0yt commented on ports/suitesparse-config/portfile.cmake:
```diff
@@ -0,0 +1,52 @@
+
+vcpkg_check_features(OUT_FEATURE_OPTIONS FEATURE_OPTIONS
+    FEATURES
+    openmp SUITESPARSE_USE_OPENMP
```
```suggestion
        openmp SUITESPARSE_USE_OPENMP
```

dg0yt commented on ports/suitesparse-graphblas/portfile.cmake:
```diff
@@ -0,0 +1,48 @@
+
+# Prevent JIT cache from being created at ~/.SuiteSparse by default. Only used during build.
+# see https://github.com/DrTimothyAldenDavis/SuiteSparse/blob/v7.8.1/GraphBLAS/cmake_modules/GraphBLAS_JIT_paths.cmake
+vcpkg_backup_env_variables(VARS GRAPHBLAS_CACHE_PATH)
```
```suggestion
```

dg0yt commented on ports/suitesparse-graphblas/portfile.cmake:
```diff
@@ -0,0 +1,48 @@
+
+vcpkg_cmake_install()
+
+vcpkg_restore_env_variables(VARS GRAPHBLAS_CACHE_PATH)
+
```
```suggestion
```
Dies with CMake script process.

dg0yt commented on ports/suitesparse-klu/portfile.cmake:
```diff
@@ -0,0 +1,48 @@
+
+vcpkg_check_features(OUT_FEATURE_OPTIONS FEATURE_OPTIONS
+    FEATURES
+    cholmod KLU_USE_CHOLMOD
```
```suggestion
        cholmod KLU_USE_CHOLMOD
```

dg0yt commented on ports/suitesparse-lagraph/portfile.cmake:
```diff
@@ -0,0 +1,42 @@
+
+vcpkg_check_features(OUT_FEATURE_OPTIONS FEATURE_OPTIONS
+    FEATURES
+    openmp  LAGRAPH_USE_OPENMP
```
```suggestion
        openmp  LAGRAPH_USE_OPENMP
```

dg0yt commented on ports/suitesparse-mongoose/vcpkg.json:
```diff
@@ -0,0 +1,24 @@
+{
+  "$comment": "Use scripts/update_suitesparse.py to update all SuiteSparse ports",
+  "name": "suitesparse-mongoose",
```
FTR there already is a mongoose port (different source).

dg0yt commented on ports/suitesparse-paru/portfile.cmake:
```diff
@@ -0,0 +1,46 @@
+
+vcpkg_check_features(OUT_FEATURE_OPTIONS FEATURE_OPTIONS
+    FEATURES
+    openmp SUITESPARSE_USE_OPENMP
```
```suggestion
        openmp SUITESPARSE_USE_OPENMP
```

dg0yt commented on ports/suitesparse-spex/portfile.cmake:
```diff
@@ -0,0 +1,45 @@
+
+vcpkg_check_features(OUT_FEATURE_OPTIONS FEATURE_OPTIONS
+    FEATURES
+    openmp SUITESPARSE_USE_OPENMP
```
```suggestion
        openmp SUITESPARSE_USE_OPENMP
```

dg0yt commented on ports/suitesparse-spqr/portfile.cmake:
```diff
@@ -0,0 +1,64 @@
+
+string(COMPARE EQUAL "${VCPKG_LIBRARY_LINKAGE}" "static" BUILD_STATIC_LIBS)
+
+set(CUDA_ENABLED OFF)
+if("cuda" IN_LIST FEATURES)
+    set(CUDA_ENABLED ON)
+    set(CUDA_ARCHITECTURES "native")
+endif()
+if("cuda-redist" IN_LIST FEATURES)
+    set(CUDA_ENABLED ON)
+    set(CUDA_ARCHITECTURES "all-major")
+endif()
```
As above.

dg0yt commented on ports/suitesparse-umfpack/portfile.cmake:
```diff
@@ -0,0 +1,45 @@
+
+vcpkg_check_features(OUT_FEATURE_OPTIONS FEATURE_OPTIONS
+    FEATURES
+    cholmod UMFPACK_USE_CHOLMOD
```
```suggestion
        cholmod UMFPACK_USE_CHOLMOD
```

dg0yt commented on ports/theia/fix-external-dependencies.patch:
```diff
@@ -200,7 +263,7 @@ index a3ea187..7e9d912 100644
 -    "dependency: Glog, please set GLOG_INCLUDE_DIR.")
 -endif (GLOG_FOUND)
-+find_package(Glog REQUIRED)
++find_dependency(Glog REQUIRED)
```
```suggestion
+find_dependency(Glog)
```

dg0yt commented on ports/theia/fix-external-dependencies.patch:
```diff
@@ -214,7 +277,7 @@ index a3ea187..7e9d912 100644
 -    "GFLAGS_INCLUDE_DIR.")
 -endif (GFLAGS_FOUND)
-+find_package(Gflags REQUIRED)
++find_dependency(Gflags REQUIRED)
```
```suggestion
+find_dependency(Gflags)
```

dg0yt commented on ports/theia/fix-external-dependencies.patch:
```diff
@@ -225,7 +288,7 @@ index a3ea187..7e9d912 100644
 -    "OPENIMAGEIO_INCLUDE_DIR & OPENIMAGEIO_LIBRARY")
 -endif (OPENIMAGEIO_FOUND)
-+find_package(OpenImageIO REQUIRED)
++find_dependency(OpenImageIO REQUIRED)
```
```suggestion
+find_dependency(OpenImageIO)
```

dg0yt commented on ports/theia/fix-external-dependencies.patch:
```diff
@@ -253,23 +316,22 @@ index a3ea187..7e9d912 100644
-+find_package(cereal REQUIRED)
+-list(APPEND THEIA_INCLUDE_DIRS ${SUITESPARSE_INCLUDE_DIRS})
++# SuiteSparse
++find_dependency(CHOLMOD REQUIRED)
++find_dependency(UMFPACK REQUIRED)
```
```suggestion
+find_dependency(CHOLMOD)
+find_dependency(UMFPACK)
```

dg0yt commented on ports/theia/fix-external-dependencies.patch:
```diff
@@ -253,23 +316,22 @@ index a3ea187..7e9d912 100644
++find_dependency(UMFPACK REQUIRED)
++list(APPEND THEIA_INCLUDE_DIRS ${CHOLMOD_INCLUDE_DIR} ${UMFPACK_INCLUDE_DIR})
++
++find_dependency(cereal REQUIRED)
```
```suggestion
+find_dependency(cereal)
```

dg0yt commented on ports/theia/fix-external-dependencies.patch:
```diff
@@ -253,23 +316,22 @@ index a3ea187..7e9d912 100644
 +list(APPEND THEIA_LIBRARIES cereal)
 +
-+find_package(Flann REQUIRED)
++find_dependency(Flann REQUIRED)
```
```suggestion
+find_dependency(Flann)
```

dg0yt commented on scripts/ci.baseline.txt:
```diff
@@ -1092,6 +1092,13 @@ stxxl:x86-windows=skip
 stxxl:x64-windows-static-md=skip
+# _Interlocked* intrinsic functions are not available on x86
+suitesparse-graphblas:x86-windows=fail
+# "ninja: error: failed recompaction: Permission denied"
```
Should be looked into. (Unresolved depenedency?)

valgur replied:
I gave that triplet another try. It still fails with the same error and I don't have any ideas what might be causing it or how to approach debugging it either.

If you're interested, here are the logs for the failed run: https://dev.azure.com/vcpkg/c1ee48cb-0df2-4ab3-8384-b1df5a79fe53/_apis/build/builds/109004/artifacts?artifactName=failure%20logs%20for%20arm64-windows&api-version=7.1&%24format=zip

dg0yt replied:
> If you're interested, here are the logs for the failed run: https://dev.azure.com/vcpkg/c1ee48cb-0df2-4ab3-8384-b1df5a79fe53/_apis/build/builds/109004/artifacts?artifactName=failure%20logs%20for%20arm64-windows&api-version=7.1&%24format=zip

Sorry, but this is gone when you don't let the build finish.

valgur replied:
Here's the failing build.ninja.log file: https://gist.github.com/valgur/a27f15e16efe93b1ed589703c3cb22cd
I still don't have any ideas why it fails with arm64 Windows triplets specifically, though. My only guess was that the file names might have been too long for Windows there, but the longest filename I found was around 170 characters, which should be ok.

dg0yt replied:
After reading https://github.com/ninja-build/ninja/issues/2493, I guess we have a cross build issue here. The trigger might be:
https://github.com/DrTimothyAldenDavis/GraphBLAS/blob/bde76fb25c95ba55f3f4070f96b59fa0caee058e/JITpackage/CMakeLists.txt#L43-L52

Now the vcpkg port probably shouldn't play this `ExternalProject` game at all. If it the package needs a host binary, it must be built in the host triplet.

dg0yt reviewed (changes requested):
I didn't check the dependency structure, but maybe `openmp` or `cuda` choices could be controlled by a single "early" port?

valgur committed:
[SuiteSparse] fix portfile.cmake indentation

valgur commented:
@dg0yt 

> I didn't check the dependency structure, but maybe `openmp` or `cuda` choices could be controlled by a single "early" port?

`suitesparse-config` would be a good place for it, I agree. Enabling OpenMP there does not transitively enable it for all other components depending on it, though. It only propagates the include dir flags in its generated CMake config. It would only help with CHOLMOD, ParU and SPEX as well, since GraphBLAS and LAGraph don't rely on suitesparse-config, so the improvement would be relatively marginal anyway.

Even more so for CUDA, which is only used by CHOLMOD and SPQR.

valgur committed:
[SuiteSparse] drop "cuda-redist" feature, keep only "cuda"

set(VCPKG_CMAKE_CONFIGURE_OPTIONS -DCMAKE_CUDA_ARCHITECTURES=...) should be added to the triplet file instead.

valgur committed:
[suitesparse-graphblas] don't back up $ENV{GRAPHBLAS_CACHE_PATH}

valgur committed:
[ceres,theia] drop REQUIRES from find_dependency()

valgur committed:
[suitesparse-cxsparse] detect complex support instead of guessing

valgur committed:
vdb

valgur committed:
[SuiteSparse] update_suitesparse.py: always update the tag in vcpkg_from_github()

To avoid having to download several versions of the source archive.

valgur committed:
[suitesparse-cholmod] need to set SUITESPARSE_USE_CUDA

valgur committed:
vdb

valgur committed:
[suitesparse-cxsparse] add a check for creal/cimg complex functions

valgur committed:
vdb

valgur committed:
[suitesparse-cxsparse] use a simpler patch

valgur committed:
vdb

valgur committed:
[suitesparse] remove leftover "cuda-redist" feature

valgur committed:
vdb

valgur commented on ports/suitesparse-graphblas/vcpkg.json:
```diff
@@ -0,0 +1,27 @@
+  "description": "SuiteSparse:GraphBLAS: graph algorithms in the language of linear algebra",
+  "homepage": "https://people.engr.tamu.edu/davis/GraphBLAS.html",
+  "license": "Apache-2.0",
+  "supports": "!android",
```
Android works perfectly fine in my tests, actually. It just requires Android API level to be set to 23 or newer for complex math functions support. The latest one is 35 and level 23 was released already back in 2015. The Vcpkg appears to be using a quite old API level.

It would be nice to encode this constraint in the port somehow, but I'm not sure how it could be done. Perhaps I'll simply remove this "support" and mark this port as expected to fail in `ci.baseline.txt`?

dg0yt replied:
Common pattern. Please add expected fail together with a comment to CI baseline. There is a lot of such comments already.

(CI was upgraded to NDK r27 recently. Time to propose the upgrade to API level 24. And then we need these comments.)

valgur committed:
[suitesparse-graphblas] add android triplets to ci.baseline.txt

valgur committed:
vdb

valgur committed:
[suitesparse-graphblas] mark arm64-windows as failing in ci.baseline.txt again

LilyWangLL commented:
New error:
```
Installing 6/54 cuda:x64-windows@10.1#13...
Building cuda:x64-windows@10.1#13...
-- Found CUDA V12.2.140
-- Installing: F:/PRreview/vcpkg/packages/cuda_x64-windows/share/cuda/copyright
-- Performing post-build validation
Stored binaries in 1 destinations in 42.9 ms.
Elapsed time to handle cuda:x64-windows: 957 ms
......
Installing 23/54 suitesparse-cholmod[core,cuda,matrixops,modify,openmp,partition,supernodal]:x64-windows@5.3.0...
Building suitesparse-cholmod[core,cuda,matrixops,modify,openmp,partition,supernodal]:x64-windows@5.3.0...
-- Using cached DrTimothyAldenDavis-SuiteSparse-v7.8.3.tar.gz.
-- Extracting source F:/PRreview/vcpkg/downloads/DrTimothyAldenDavis-SuiteSparse-v7.8.3.tar.gz
-- Using source at F:/PRreview/vcpkg/buildtrees/suitesparse-cholmod/src/v7.8.3-ee8b79bb01.clean
-- Configuring x64-windows-dbg
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:127 (message):
    Command failed: F:/PRreview/vcpkg/downloads/tools/cmake-3.30.1-windows/cmake-3.30.1-windows-i386/bin/cmake.exe F:/PRreview/vcpkg/buildtrees/suitesparse-cholmod/src/v7.8.3-ee8b79bb01.clean/CHOLMOD -G Ninja -DCMAKE_BUILD_TYPE=Debug -DCMAKE_INSTALL_PREFIX=F:/PRreview/vcpkg/packages/suitesparse-cholmod_x64-windows/debug -DFETCHCONTENT_FULLY_DISCONNECTED=ON -DBUILD_STATIC_LIBS=0 -DCHOLMOD_GPL=ON -DSUITESPARSE_USE_STRICT=ON -DSUITESPARSE_USE_FORTRAN=OFF -DSUITESPARSE_DEMOS=OFF -DSUITESPARSE_USE_64BIT_BLAS=1 -DCHOLMOD_USE_CUDA=ON -DSUITESPARSE_USE_CUDA=ON -DCHOLMOD_MATRIXOPS=ON -DCHOLMOD_MODIFY=ON -DCHOLMOD_PARTITION=ON -DCHOLMOD_SUPERNODAL=ON -DCHOLMOD_USE_OPENMP=ON -DCMAKE_MAKE_PROGRAM=F:/PRreview/vcpkg/downloads/tools/ninja/1.10.2-windows/ninja.exe -DBUILD_SHARED_LIBS=ON -DVCPKG_CHAINLOAD_TOOLCHAIN_FILE=F:/PRreview/vcpkg/scripts/toolchains/windows.cmake -DVCPKG_TARGET_TRIPLET=x64-windows -DVCPKG_SET_CHARSET_FLAG=ON -DVCPKG_PLATFORM_TOOLSET=v143 -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON -DCMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY=ON -DCMAKE_FIND_PACKAGE_NO_SYSTEM_PACKAGE_REGISTRY=ON -DCMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP=TRUE -DCMAKE_VERBOSE_MAKEFILE=ON -DVCPKG_APPLOCAL_DEPS=OFF -DCMAKE_TOOLCHAIN_FILE=F:/PRreview/vcpkg/scripts/buildsystems/vcpkg.cmake -DCMAKE_ERROR_ON_ABSOLUTE_INSTALL_DESTINATION=ON -DVCPKG_CXX_FLAGS= -DVCPKG_CXX_FLAGS_RELEASE= -DVCPKG_CXX_FLAGS_DEBUG= -DVCPKG_C_FLAGS= -DVCPKG_C_FLAGS_RELEASE= -DVCPKG_C_FLAGS_DEBUG= -DVCPKG_CRT_LINKAGE=dynamic -DVCPKG_LINKER_FLAGS= -DVCPKG_LINKER_FLAGS_RELEASE= -DVCPKG_LINKER_FLAGS_DEBUG= -DVCPKG_TARGET_ARCHITECTURE=x64 -DCMAKE_INSTALL_LIBDIR:STRING=lib -DCMAKE_INSTALL_BINDIR:STRING=bin -D_VCPKG_ROOT_DIR=F:/PRreview/vcpkg -D_VCPKG_INSTALLED_DIR=F:/PRreview/vcpkg/installed -DVCPKG_MANIFEST_INSTALL=OFF
    Working Directory: F:/PRreview/vcpkg/buildtrees/suitesparse-cholmod/x64-windows-dbg
    Error code: 1
    See logs for more information:
      F:\PRreview\vcpkg\buildtrees\suitesparse-cholmod\config-x64-windows-dbg-CMakeCache.txt.log
      F:\PRreview\vcpkg\buildtrees\suitesparse-cholmod\config-x64-windows-dbg-CMakeConfigureLog.yaml.log
      F:\PRreview\vcpkg\buildtrees\suitesparse-cholmod\config-x64-windows-dbg-out.log
      F:\PRreview\vcpkg\buildtrees\suitesparse-cholmod\config-x64-windows-dbg-err.log
```
The error message from **F:\PRreview\vcpkg\buildtrees\suitesparse-cholmod\config-x64-windows-dbg-err.log**:
```
CMake Error at F:/PRreview/vcpkg/buildtrees/suitesparse-cholmod/src/v7.8.3-ee8b79bb01.clean/SuiteSparse_config/cmake_modules/SuiteSparsePolicy.cmake:367 (message):
  CUDA required for SuiteSparse but not found
Call Stack (most recent call first):
  CMakeLists.txt:41 (include)
```
CUDA has been installed.

valgur committed:
[SuiteSparse] don't override CMAKE_CUDA_ARCHITECTURES

Also improve the error messages for CUDA detection in CMake.

valgur committed:
vdb

valgur commented:
@LilyWangLL I now noticed that the project was always overriding CMAKE_CUDA_ARCHITECTURES. I fixed that and also made the CUDA detection error reporting more immediate and hopefully clearer. Can you try again?

LilyWangLL commented:
The new error when install `suitesparse-spqr[core,cuda]`:
```
CMake Error at F:/PRreview/vcpkg/downloads/tools/cmake-3.30.1-windows/cmake-3.30.1-windows-i386/share/cmake-3.30/Modules/Internal/CMakeCUDAArchitecturesValidate.cmake:7 (message):
  CMAKE_CUDA_ARCHITECTURES must be non-empty if set.
Call Stack (most recent call first):
  F:/PRreview/vcpkg/downloads/tools/cmake-3.30.1-windows/cmake-3.30.1-windows-i386/share/cmake-3.30/Modules/CMakeDetermineCUDACompiler.cmake:112 (cmake_cuda_architectures_validate)
  F:/PRreview/vcpkg/buildtrees/suitesparse-spqr/src/v7.8.3-371eca9f9f.clean/SuiteSparse_config/cmake_modules/SuiteSparsePolicy.cmake:324 (enable_language)
  CMakeLists.txt:41 (include)
```
This error disappeared after I add `set(CMAKE_CUDA_ARCHITECTURES "native")` in _F:/PRreview/vcpkg/buildtrees/suitesparse-spqr/src/v7.8.3-371eca9f9f.clean/SuiteSparse_config/cmake_modules/SuiteSparsePolicy.cmake_.

LilyWangLL commented:
And there are new LINK errors when install `suitesparse-cholmod[core,cuda,matrixops,modify,openmp,partition,supernodal]:x64-windows-static`:
```
[224/228] C:\Windows\system32\cmd.exe /C "cd . && F:\PRreview\vcpkg\downloads\tools\cmake-3.30.1-windows\cmake-3.30.1-windows-i386\bin\cmake.exe -E vs_link_exe --intdir=CMakeFiles\cholmod_di_demo.dir --rc=C:\PROGRA~2\WI3CF2~1\10\bin\100261~1.0\x64\rc.exe --mt=C:\PROGRA~2\WI3CF2~1\10\bin\100261~1.0\x64\mt.exe --manifests  -- C:\PROGRA~2\MICROS~2\2019\ENTERP~1\VC\Tools\MSVC\1429~1.301\bin\Hostx64\x64\link.exe  CMakeFiles\cholmod_di_demo.dir\Demo\cholmod_di_demo.c.obj CMakeFiles\cholmod_di_demo.dir\cmake_device_link.obj  /out:cholmod_di_demo.exe /implib:cholmod_di_demo.lib /pdb:cholmod_di_demo.pdb /version:0.0 /machine:x64 /nologo    /debug /INCREMENTAL  cholmod_static.lib  F:\PRreview\vcpkg\installed\x64-windows-static\debug\lib\suitesparseconfig_static.lib  F:\PRreview\vcpkg\installed\x64-windows-static\debug\lib\amd_static.lib  F:\PRreview\vcpkg\installed\x64-windows-static\debug\lib\colamd_static.lib  F:\PRreview\vcpkg\installed\x64-windows-static\debug\lib\camd_static.lib  F:\PRreview\vcpkg\installed\x64-windows-static\debug\lib\ccolamd_static.lib  F:\PRreview\vcpkg\installed\x64-windows-static\debug\lib\suitesparseconfig_static.lib  F:\PRreview\vcpkg\installed\x64-windows-static\debug\lib\lapack.lib  F:\PRreview\vcpkg\installed\x64-windows-static\debug\lib\blas.lib  F:\PRreview\vcpkg\installed\x64-windows-static\debug\lib\cblas.lib  F:\PRreview\vcpkg\installed\x64-windows-static\debug\lib\blas.lib  F:\PRreview\vcpkg\installed\x64-windows-static\debug\lib\cblas.lib  "C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v12.2\lib\x64\nvrtc.lib"  "C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v12.2\lib\x64\nvJitLink.lib"  "C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v12.2\lib\x64\cudart_static.lib"  "C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v12.2\lib\x64\cublas.lib"  "C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v12.2\lib\x64\cublasLt.lib"  cudadevrt.lib  cudart_static.lib  kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib -LIBPATH:"C:/Program Files/NVIDIA GPU Computing Toolkit/CUDA/v12.2/lib/x64" && cd ."
FAILED: cholmod_di_demo.exe 
C:\Windows\system32\cmd.exe /C "cd . && F:\PRreview\vcpkg\downloads\tools\cmake-3.30.1-windows\cmake-3.30.1-windows-i386\bin\cmake.exe -E vs_link_exe --intdir=CMakeFiles\cholmod_di_demo.dir --rc=C:\PROGRA~2\WI3CF2~1\10\bin\100261~1.0\x64\rc.exe --mt=C:\PROGRA~2\WI3CF2~1\10\bin\100261~1.0\x64\mt.exe --manifests  -- C:\PROGRA~2\MICROS~2\2019\ENTERP~1\VC\Tools\MSVC\1429~1.301\bin\Hostx64\x64\link.exe  CMakeFiles\cholmod_di_demo.dir\Demo\cholmod_di_demo.c.obj CMakeFiles\cholmod_di_demo.dir\cmake_device_link.obj  /out:cholmod_di_demo.exe /implib:cholmod_di_demo.lib /pdb:cholmod_di_demo.pdb /version:0.0 /machine:x64 /nologo    /debug /INCREMENTAL  cholmod_static.lib  F:\PRreview\vcpkg\installed\x64-windows-static\debug\lib\suitesparseconfig_static.lib  F:\PRreview\vcpkg\installed\x64-windows-static\debug\lib\amd_static.lib  F:\PRreview\vcpkg\installed\x64-windows-static\debug\lib\colamd_static.lib  F:\PRreview\vcpkg\installed\x64-windows-static\debug\lib\camd_static.lib  F:\PRreview\vcpkg\installed\x64-windows-static\debug\lib\ccolamd_static.lib  F:\PRreview\vcpkg\installed\x64-windows-static\debug\lib\suitesparseconfig_static.lib  F:\PRreview\vcpkg\installed\x64-windows-static\debug\lib\lapack.lib  F:\PRreview\vcpkg\installed\x64-windows-static\debug\lib\blas.lib  F:\PRreview\vcpkg\installed\x64-windows-static\debug\lib\cblas.lib  F:\PRreview\vcpkg\installed\x64-windows-static\debug\lib\blas.lib  F:\PRreview\vcpkg\installed\x64-windows-static\debug\lib\cblas.lib  "C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v12.2\lib\x64\nvrtc.lib"  "C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v12.2\lib\x64\nvJitLink.lib"  "C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v12.2\lib\x64\cudart_static.lib"  "C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v12.2\lib\x64\cublas.lib"  "C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v12.2\lib\x64\cublasLt.lib"  cudadevrt.lib  cudart_static.lib  kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib -LIBPATH:"C:/Program Files/NVIDIA GPU Computing Toolkit/CUDA/v12.2/lib/x64" && cd ."
LINK Pass 1: command "C:\PROGRA~2\MICROS~2\2019\ENTERP~1\VC\Tools\MSVC\1429~1.301\bin\Hostx64\x64\link.exe CMakeFiles\cholmod_di_demo.dir\Demo\cholmod_di_demo.c.obj CMakeFiles\cholmod_di_demo.dir\cmake_device_link.obj /out:cholmod_di_demo.exe /implib:cholmod_di_demo.lib /pdb:cholmod_di_demo.pdb /version:0.0 /machine:x64 /nologo /debug /INCREMENTAL cholmod_static.lib F:\PRreview\vcpkg\installed\x64-windows-static\debug\lib\suitesparseconfig_static.lib F:\PRreview\vcpkg\installed\x64-windows-static\debug\lib\amd_static.lib F:\PRreview\vcpkg\installed\x64-windows-static\debug\lib\colamd_static.lib F:\PRreview\vcpkg\installed\x64-windows-static\debug\lib\camd_static.lib F:\PRreview\vcpkg\installed\x64-windows-static\debug\lib\ccolamd_static.lib F:\PRreview\vcpkg\installed\x64-windows-static\debug\lib\suitesparseconfig_static.lib F:\PRreview\vcpkg\installed\x64-windows-static\debug\lib\lapack.lib F:\PRreview\vcpkg\installed\x64-windows-static\debug\lib\blas.lib F:\PRreview\vcpkg\installed\x64-windows-static\debug\lib\cblas.lib F:\PRreview\vcpkg\installed\x64-windows-static\debug\lib\blas.lib F:\PRreview\vcpkg\installed\x64-windows-static\debug\lib\cblas.lib C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v12.2\lib\x64\nvrtc.lib C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v12.2\lib\x64\nvJitLink.lib C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v12.2\lib\x64\cudart_static.lib C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v12.2\lib\x64\cublas.lib C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v12.2\lib\x64\cublasLt.lib cudadevrt.lib cudart_static.lib kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib -LIBPATH:C:/Program Files/NVIDIA GPU Computing Toolkit/CUDA/v12.2/lib/x64 /MANIFEST /MANIFESTFILE:CMakeFiles\cholmod_di_demo.dir/intermediate.manifest CMakeFiles\cholmod_di_demo.dir/manifest.res" failed (exit code 1120) with the following output:
   Creating library cholmod_di_demo.lib and object cholmod_di_demo.exp
LINK : warning LNK4098: defaultlib 'LIBCMT' conflicts with use of other libs; use /NODEFAULTLIB:library
cholmod_static.lib(cholmod_super_numeric.c.obj) : error LNK2019: unresolved external symbol dtrsm__ referenced in function rd_cholmod_super_numeric_worker
cholmod_static.lib(cholmod_super_solve.c.obj) : error LNK2001: unresolved external symbol dtrsm__
cholmod_static.lib(cholmod_super_numeric.c.obj) : error LNK2019: unresolved external symbol strsm__ referenced in function rs_cholmod_super_numeric_worker
cholmod_static.lib(cholmod_super_solve.c.obj) : error LNK2001: unresolved external symbol strsm__
cholmod_static.lib(cholmod_super_numeric.c.obj) : error LNK2019: unresolved external symbol ztrsm__ referenced in function cd_cholmod_super_numeric_worker
cholmod_static.lib(cholmod_super_solve.c.obj) : error LNK2001: unresolved external symbol ztrsm__
cholmod_static.lib(cholmod_super_numeric.c.obj) : error LNK2019: unresolved external symbol ctrsm__ referenced in function cs_cholmod_super_numeric_worker
cholmod_static.lib(cholmod_super_solve.c.obj) : error LNK2001: unresolved external symbol ctrsm__
cholmod_static.lib(cholmod_super_numeric.c.obj) : error LNK2019: unresolved external symbol dgemm__ referenced in function rd_cholmod_super_numeric_worker
cholmod_static.lib(cholmod_super_solve.c.obj) : error LNK2001: unresolved external symbol dgemm__
cholmod_static.lib(cholmod_super_numeric.c.obj) : error LNK2019: unresolved external symbol sgemm__ referenced in function rs_cholmod_super_numeric_worker
cholmod_static.lib(cholmod_super_solve.c.obj) : error LNK2001: unresolved external symbol sgemm__
cholmod_static.lib(cholmod_super_numeric.c.obj) : error LNK2019: unresolved external symbol zgemm__ referenced in function cd_cholmod_super_numeric_worker
cholmod_static.lib(cholmod_super_solve.c.obj) : error LNK2001: unresolved external symbol zgemm__
cholmod_static.lib(cholmod_super_numeric.c.obj) : error LNK2019: unresolved external symbol cgemm__ referenced in function cs_cholmod_super_numeric_worker
cholmod_static.lib(cholmod_super_solve.c.obj) : error LNK2001: unresolved external symbol cgemm__
cholmod_static.lib(cholmod_super_numeric.c.obj) : error LNK2019: unresolved external symbol dsyrk__ referenced in function rd_cholmod_super_numeric_worker
cholmod_static.lib(cholmod_super_numeric.c.obj) : error LNK2019: unresolved external symbol ssyrk__ referenced in function rs_cholmod_super_numeric_worker
cholmod_static.lib(cholmod_super_numeric.c.obj) : error LNK2019: unresolved external symbol zherk__ referenced in function cd_cholmod_super_numeric_worker
cholmod_static.lib(cholmod_super_numeric.c.obj) : error LNK2019: unresolved external symbol cherk__ referenced in function cs_cholmod_super_numeric_worker
cholmod_static.lib(cholmod_super_numeric.c.obj) : error LNK2019: unresolved external symbol dpotrf__ referenced in function rd_cholmod_super_numeric_worker
cholmod_static.lib(cholmod_super_numeric.c.obj) : error LNK2019: unresolved external symbol spotrf__ referenced in function rs_cholmod_super_numeric_worker
cholmod_static.lib(cholmod_super_numeric.c.obj) : error LNK2019: unresolved external symbol zpotrf__ referenced in function cd_cholmod_super_numeric_worker
cholmod_static.lib(cholmod_super_numeric.c.obj) : error LNK2019: unresolved external symbol cpotrf__ referenced in function cs_cholmod_super_numeric_worker
cholmod_static.lib(cholmod_super_solve.c.obj) : error LNK2019: unresolved external symbol dgemv__ referenced in function rd_cholmod_super_lsolve_worker
cholmod_static.lib(cholmod_super_solve.c.obj) : error LNK2019: unresolved external symbol sgemv__ referenced in function rs_cholmod_super_lsolve_worker
cholmod_static.lib(cholmod_super_solve.c.obj) : error LNK2019: unresolved external symbol zgemv__ referenced in function cd_cholmod_super_lsolve_worker
cholmod_static.lib(cholmod_super_solve.c.obj) : error LNK2019: unresolved external symbol cgemv__ referenced in function cs_cholmod_super_lsolve_worker
cholmod_static.lib(cholmod_super_solve.c.obj) : error LNK2019: unresolved external symbol dtrsv__ referenced in function rd_cholmod_super_lsolve_worker
cholmod_static.lib(cholmod_super_solve.c.obj) : error LNK2019: unresolved external symbol strsv__ referenced in function rs_cholmod_super_lsolve_worker
cholmod_static.lib(cholmod_super_solve.c.obj) : error LNK2019: unresolved external symbol ztrsv__ referenced in function cd_cholmod_super_lsolve_worker
cholmod_static.lib(cholmod_super_solve.c.obj) : error LNK2019: unresolved external symbol ctrsv__ referenced in function cs_cholmod_super_lsolve_worker
cholmod_di_demo.exe : fatal error LNK1120: 24 unresolved externals
```

valgur commented:
> The new error when install `suitesparse-spqr[core,cuda]`:
> 
> ```
> CMake Error at F:/PRreview/vcpkg/downloads/tools/cmake-3.30.1-windows/cmake-3.30.1-windows-i386/share/cmake-3.30/Modules/Internal/CMakeCUDAArchitecturesValidate.cmake:7 (message):
>   CMAKE_CUDA_ARCHITECTURES must be non-empty if set.
> Call Stack (most recent call first):
>   F:/PRreview/vcpkg/downloads/tools/cmake-3.30.1-windows/cmake-3.30.1-windows-i386/share/cmake-3.30/Modules/CMakeDetermineCUDACompiler.cmake:112 (cmake_cuda_architectures_validate)
>   F:/PRreview/vcpkg/buildtrees/suitesparse-spqr/src/v7.8.3-371eca9f9f.clean/SuiteSparse_config/cmake_modules/SuiteSparsePolicy.cmake:324 (enable_language)
>   CMakeLists.txt:41 (include)
> ```
> 
> This error disappeared after I add `set(CMAKE_CUDA_ARCHITECTURES "native")` in _F:/PRreview/vcpkg/buildtrees/suitesparse-spqr/src/v7.8.3-371eca9f9f.clean/SuiteSparse_config/cmake_modules/SuiteSparsePolicy.cmake_.

That was by design this and I would say that is something that absolutely be a part of the triplet file when using CUDA in the first place. E.g. `set(VCPKG_CMAKE_CONFIGURE_OPTIONS -DCMAKE_CUDA_ARCHITECTURES=native)`.

I would definitely want to leave the user the option of configuring the set of CUDA architectures to build for but I don't see a good way to also include a default set of architectures without overriding the user's choice. I'm open to suggestions!

See also https://github.com/microsoft/vcpkg/pull/41929#discussion_r1826949529

valgur commented:
> And there are new LINK errors when install `suitesparse-cholmod[core,cuda,matrixops,modify,openmp,partition,supernodal]:x64-windows-static`:
> 
> ```
> [224/228] C:\Windows\system32\cmd.exe /C "cd . && F:\PRreview\vcpkg\downloads\tools\cmake-3.30.1-windows\cmake-3.30.1-windows-i386\bin\cmake.exe -E vs_link_exe --intdir=CMakeFiles\cholmod_di_demo.dir --rc=C:\PROGRA~2\WI3CF2~1\10\bin\100261~1.0\x64\rc.exe --mt=C:\PROGRA~2\WI3CF2~1\10\bin\100261~1.0\x64\mt.exe --manifests  -- C:\PROGRA~2\MICROS~2\2019\ENTERP~1\VC\Tools\MSVC\1429~1.301\bin\Hostx64\x64\link.exe  CMakeFiles\cholmod_di_demo.dir\Demo\cholmod_di_demo.c.obj CMakeFiles\cholmod_di_demo.dir\cmake_device_link.obj  /out:cholmod_di_demo.exe /implib:cholmod_di_demo.lib /pdb:cholmod_di_demo.pdb /version:0.0 /machine:x64 /nologo    /debug /INCREMENTAL  cholmod_static.lib  F:\PRreview\vcpkg\installed\x64-windows-static\debug\lib\suitesparseconfig_static.lib  F:\PRreview\vcpkg\installed\x64-windows-static\debug\lib\amd_static.lib  F:\PRreview\vcpkg\installed\x64-windows-static\debug\lib\colamd_static.lib  F:\PRreview\vcpkg\installed\x64-windows-static\debug\lib\camd_static.lib  F:\PRreview\vcpkg\installed\x64-windows-static\debug\lib\ccolamd_static.lib  F:\PRreview\vcpkg\installed\x64-windows-static\debug\lib\suitesparseconfig_static.lib  F:\PRreview\vcpkg\installed\x64-windows-static\debug\lib\lapack.lib  F:\PRreview\vcpkg\installed\x64-windows-static\debug\lib\blas.lib  F:\PRreview\vcpkg\installed\x64-windows-static\debug\lib\cblas.lib  F:\PRreview\vcpkg\installed\x64-windows-static\debug\lib\blas.lib  F:\PRreview\vcpkg\installed\x64-windows-static\debug\lib\cblas.lib  "C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v12.2\lib\x64\nvrtc.lib"  "C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v12.2\lib\x64\nvJitLink.lib"  "C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v12.2\lib\x64\cudart_static.lib"  "C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v12.2\lib\x64\cublas.lib"  "C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v12.2\lib\x64\cublasLt.lib"  cudadevrt.lib  cudart_static.lib  kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib -LIBPATH:"C:/Program Files/NVIDIA GPU Computing Toolkit/CUDA/v12.2/lib/x64" && cd ."
> FAILED: cholmod_di_demo.exe 
> C:\Windows\system32\cmd.exe /C "cd . && F:\PRreview\vcpkg\downloads\tools\cmake-3.30.1-windows\cmake-3.30.1-windows-i386\bin\cmake.exe -E vs_link_exe --intdir=CMakeFiles\cholmod_di_demo.dir --rc=C:\PROGRA~2\WI3CF2~1\10\bin\100261~1.0\x64\rc.exe --mt=C:\PROGRA~2\WI3CF2~1\10\bin\100261~1.0\x64\mt.exe --manifests  -- C:\PROGRA~2\MICROS~2\2019\ENTERP~1\VC\Tools\MSVC\1429~1.301\bin\Hostx64\x64\link.exe  CMakeFiles\cholmod_di_demo.dir\Demo\cholmod_di_demo.c.obj CMakeFiles\cholmod_di_demo.dir\cmake_device_link.obj  /out:cholmod_di_demo.exe /implib:cholmod_di_demo.lib /pdb:cholmod_di_demo.pdb /version:0.0 /machine:x64 /nologo    /debug /INCREMENTAL  cholmod_static.lib  F:\PRreview\vcpkg\installed\x64-windows-static\debug\lib\suitesparseconfig_static.lib  F:\PRreview\vcpkg\installed\x64-windows-static\debug\lib\amd_static.lib  F:\PRreview\vcpkg\installed\x64-windows-static\debug\lib\colamd_static.lib  F:\PRreview\vcpkg\installed\x64-windows-static\debug\lib\camd_static.lib  F:\PRreview\vcpkg\installed\x64-windows-static\debug\lib\ccolamd_static.lib  F:\PRreview\vcpkg\installed\x64-windows-static\debug\lib\suitesparseconfig_static.lib  F:\PRreview\vcpkg\installed\x64-windows-static\debug\lib\lapack.lib  F:\PRreview\vcpkg\installed\x64-windows-static\debug\lib\blas.lib  F:\PRreview\vcpkg\installed\x64-windows-static\debug\lib\cblas.lib  F:\PRreview\vcpkg\installed\x64-windows-static\debug\lib\blas.lib  F:\PRreview\vcpkg\installed\x64-windows-static\debug\lib\cblas.lib  "C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v12.2\lib\x64\nvrtc.lib"  "C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v12.2\lib\x64\nvJitLink.lib"  "C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v12.2\lib\x64\cudart_static.lib"  "C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v12.2\lib\x64\cublas.lib"  "C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v12.2\lib\x64\cublasLt.lib"  cudadevrt.lib  cudart_static.lib  kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib -LIBPATH:"C:/Program Files/NVIDIA GPU Computing Toolkit/CUDA/v12.2/lib/x64" && cd ."
> LINK Pass 1: command "C:\PROGRA~2\MICROS~2\2019\ENTERP~1\VC\Tools\MSVC\1429~1.301\bin\Hostx64\x64\link.exe CMakeFiles\cholmod_di_demo.dir\Demo\cholmod_di_demo.c.obj CMakeFiles\cholmod_di_demo.dir\cmake_device_link.obj /out:cholmod_di_demo.exe /implib:cholmod_di_demo.lib /pdb:cholmod_di_demo.pdb /version:0.0 /machine:x64 /nologo /debug /INCREMENTAL cholmod_static.lib F:\PRreview\vcpkg\installed\x64-windows-static\debug\lib\suitesparseconfig_static.lib F:\PRreview\vcpkg\installed\x64-windows-static\debug\lib\amd_static.lib F:\PRreview\vcpkg\installed\x64-windows-static\debug\lib\colamd_static.lib F:\PRreview\vcpkg\installed\x64-windows-static\debug\lib\camd_static.lib F:\PRreview\vcpkg\installed\x64-windows-static\debug\lib\ccolamd_static.lib F:\PRreview\vcpkg\installed\x64-windows-static\debug\lib\suitesparseconfig_static.lib F:\PRreview\vcpkg\installed\x64-windows-static\debug\lib\lapack.lib F:\PRreview\vcpkg\installed\x64-windows-static\debug\lib\blas.lib F:\PRreview\vcpkg\installed\x64-windows-static\debug\lib\cblas.lib F:\PRreview\vcpkg\installed\x64-windows-static\debug\lib\blas.lib F:\PRreview\vcpkg\installed\x64-windows-static\debug\lib\cblas.lib C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v12.2\lib\x64\nvrtc.lib C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v12.2\lib\x64\nvJitLink.lib C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v12.2\lib\x64\cudart_static.lib C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v12.2\lib\x64\cublas.lib C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v12.2\lib\x64\cublasLt.lib cudadevrt.lib cudart_static.lib kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib -LIBPATH:C:/Program Files/NVIDIA GPU Computing Toolkit/CUDA/v12.2/lib/x64 /MANIFEST /MANIFESTFILE:CMakeFiles\cholmod_di_demo.dir/intermediate.manifest CMakeFiles\cholmod_di_demo.dir/manifest.res" failed (exit code 1120) with the following output:
>    Creating library cholmod_di_demo.lib and object cholmod_di_demo.exp
> LINK : warning LNK4098: defaultlib 'LIBCMT' conflicts with use of other libs; use /NODEFAULTLIB:library
> cholmod_static.lib(cholmod_super_numeric.c.obj) : error LNK2019: unresolved external symbol dtrsm__ referenced in function rd_cholmod_super_numeric_worker
> cholmod_static.lib(cholmod_super_solve.c.obj) : error LNK2001: unresolved external symbol dtrsm__
> cholmod_static.lib(cholmod_super_numeric.c.obj) : error LNK2019: unresolved external symbol strsm__ referenced in function rs_cholmod_super_numeric_worker
> cholmod_static.lib(cholmod_super_solve.c.obj) : error LNK2001: unresolved external symbol strsm__
> cholmod_static.lib(cholmod_super_numeric.c.obj) : error LNK2019: unresolved external symbol ztrsm__ referenced in function cd_cholmod_super_numeric_worker
> cholmod_static.lib(cholmod_super_solve.c.obj) : error LNK2001: unresolved external symbol ztrsm__
> cholmod_static.lib(cholmod_super_numeric.c.obj) : error LNK2019: unresolved external symbol ctrsm__ referenced in function cs_cholmod_super_numeric_worker
> cholmod_static.lib(cholmod_super_solve.c.obj) : error LNK2001: unresolved external symbol ctrsm__
> cholmod_static.lib(cholmod_super_numeric.c.obj) : error LNK2019: unresolved external symbol dgemm__ referenced in function rd_cholmod_super_numeric_worker
> cholmod_static.lib(cholmod_super_solve.c.obj) : error LNK2001: unresolved external symbol dgemm__
> cholmod_static.lib(cholmod_super_numeric.c.obj) : error LNK2019: unresolved external symbol sgemm__ referenced in function rs_cholmod_super_numeric_worker
> cholmod_static.lib(cholmod_super_solve.c.obj) : error LNK2001: unresolved external symbol sgemm__
> cholmod_static.lib(cholmod_super_numeric.c.obj) : error LNK2019: unresolved external symbol zgemm__ referenced in function cd_cholmod_super_numeric_worker
> cholmod_static.lib(cholmod_super_solve.c.obj) : error LNK2001: unresolved external symbol zgemm__
> cholmod_static.lib(cholmod_super_numeric.c.obj) : error LNK2019: unresolved external symbol cgemm__ referenced in function cs_cholmod_super_numeric_worker
> cholmod_static.lib(cholmod_super_solve.c.obj) : error LNK2001: unresolved external symbol cgemm__
> cholmod_static.lib(cholmod_super_numeric.c.obj) : error LNK2019: unresolved external symbol dsyrk__ referenced in function rd_cholmod_super_numeric_worker
> cholmod_static.lib(cholmod_super_numeric.c.obj) : error LNK2019: unresolved external symbol ssyrk__ referenced in function rs_cholmod_super_numeric_worker
> cholmod_static.lib(cholmod_super_numeric.c.obj) : error LNK2019: unresolved external symbol zherk__ referenced in function cd_cholmod_super_numeric_worker
> cholmod_static.lib(cholmod_super_numeric.c.obj) : error LNK2019: unresolved external symbol cherk__ referenced in function cs_cholmod_super_numeric_worker
> cholmod_static.lib(cholmod_super_numeric.c.obj) : error LNK2019: unresolved external symbol dpotrf__ referenced in function rd_cholmod_super_numeric_worker
> cholmod_static.lib(cholmod_super_numeric.c.obj) : error LNK2019: unresolved external symbol spotrf__ referenced in function rs_cholmod_super_numeric_worker
> cholmod_static.lib(cholmod_super_numeric.c.obj) : error LNK2019: unresolved external symbol zpotrf__ referenced in function cd_cholmod_super_numeric_worker
> cholmod_static.lib(cholmod_super_numeric.c.obj) : error LNK2019: unresolved external symbol cpotrf__ referenced in function cs_cholmod_super_numeric_worker
> cholmod_static.lib(cholmod_super_solve.c.obj) : error LNK2019: unresolved external symbol dgemv__ referenced in function rd_cholmod_super_lsolve_worker
> cholmod_static.lib(cholmod_super_solve.c.obj) : error LNK2019: unresolved external symbol sgemv__ referenced in function rs_cholmod_super_lsolve_worker
> cholmod_static.lib(cholmod_super_solve.c.obj) : error LNK2019: unresolved external symbol zgemv__ referenced in function cd_cholmod_super_lsolve_worker
> cholmod_static.lib(cholmod_super_solve.c.obj) : error LNK2019: unresolved external symbol cgemv__ referenced in function cs_cholmod_super_lsolve_worker
> cholmod_static.lib(cholmod_super_solve.c.obj) : error LNK2019: unresolved external symbol dtrsv__ referenced in function rd_cholmod_super_lsolve_worker
> cholmod_static.lib(cholmod_super_solve.c.obj) : error LNK2019: unresolved external symbol strsv__ referenced in function rs_cholmod_super_lsolve_worker
> cholmod_static.lib(cholmod_super_solve.c.obj) : error LNK2019: unresolved external symbol ztrsv__ referenced in function cd_cholmod_super_lsolve_worker
> cholmod_static.lib(cholmod_super_solve.c.obj) : error LNK2019: unresolved external symbol ctrsv__ referenced in function cs_cholmod_super_lsolve_worker
> cholmod_di_demo.exe : fatal error LNK1120: 24 unresolved externals
> ```

I'm afraid I don't have the capacity to deal with this edge case at the moment. Would it be possible to move forward with this known failure case?

sarlinpe commented:
Any chance that this could go through? We'd love to ship in COLMAP the new GPU features of Ceres 2.2/2.3.

valgur committed:
trigger CI

ahojnnes commented:
Independent of the updates to suitesparse, I am updating Ceres to 2.2.0 here. It should be rather straightforward to modify the patches in the Ceres port to consume the updated SuiteSparse port here.

MonicaLiu0311 commented:
Please resolve conflicts first.

valgur committed:
Merge remote-tracking branch 'origin/master' into update/suitesparse

# Conflicts:
#	versions/t-/theia.json

valgur committed:
vdb

valgur committed:
[suitesparse-graphblas] add arm64-windows-static-md=fail to ci.baseline.txt

valgur committed:
Merge remote-tracking branch 'origin/master' into update/suitesparse

LilyWangLL commented:
> > And there are new LINK errors when install `suitesparse-cholmod[core,cuda,matrixops,modify,openmp,partition,supernodal]:x64-windows-static`:
> > ```
> > [224/228] C:\Windows\system32\cmd.exe /C "cd . && F:\PRreview\vcpkg\downloads\tools\cmake-3.30.1-windows\cmake-3.30.1-windows-i386\bin\cmake.exe -E vs_link_exe --intdir=CMakeFiles\cholmod_di_demo.dir --rc=C:\PROGRA~2\WI3CF2~1\10\bin\100261~1.0\x64\rc.exe --mt=C:\PROGRA~2\WI3CF2~1\10\bin\100261~1.0\x64\mt.exe --manifests  -- C:\PROGRA~2\MICROS~2\2019\ENTERP~1\VC\Tools\MSVC\1429~1.301\bin\Hostx64\x64\link.exe  CMakeFiles\cholmod_di_demo.dir\Demo\cholmod_di_demo.c.obj CMakeFiles\cholmod_di_demo.dir\cmake_device_link.obj  /out:cholmod_di_demo.exe /implib:cholmod_di_demo.lib /pdb:cholmod_di_demo.pdb /version:0.0 /machine:x64 /nologo    /debug /INCREMENTAL  cholmod_static.lib  F:\PRreview\vcpkg\installed\x64-windows-static\debug\lib\suitesparseconfig_static.lib  F:\PRreview\vcpkg\installed\x64-windows-static\debug\lib\amd_static.lib  F:\PRreview\vcpkg\installed\x64-windows-static\debug\lib\colamd_static.lib  F:\PRreview\vcpkg\installed\x64-windows-static\debug\lib\camd_static.lib  F:\PRreview\vcpkg\installed\x64-windows-static\debug\lib\ccolamd_static.lib  F:\PRreview\vcpkg\installed\x64-windows-static\debug\lib\suitesparseconfig_static.lib  F:\PRreview\vcpkg\installed\x64-windows-static\debug\lib\lapack.lib  F:\PRreview\vcpkg\installed\x64-windows-static\debug\lib\blas.lib  F:\PRreview\vcpkg\installed\x64-windows-static\debug\lib\cblas.lib  F:\PRreview\vcpkg\installed\x64-windows-static\debug\lib\blas.lib  F:\PRreview\vcpkg\installed\x64-windows-static\debug\lib\cblas.lib  "C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v12.2\lib\x64\nvrtc.lib"  "C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v12.2\lib\x64\nvJitLink.lib"  "C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v12.2\lib\x64\cudart_static.lib"  "C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v12.2\lib\x64\cublas.lib"  "C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v12.2\lib\x64\cublasLt.lib"  cudadevrt.lib  cudart_static.lib  kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib -LIBPATH:"C:/Program Files/NVIDIA GPU Computing Toolkit/CUDA/v12.2/lib/x64" && cd ."
> > FAILED: cholmod_di_demo.exe 
> > C:\Windows\system32\cmd.exe /C "cd . && F:\PRreview\vcpkg\downloads\tools\cmake-3.30.1-windows\cmake-3.30.1-windows-i386\bin\cmake.exe -E vs_link_exe --intdir=CMakeFiles\cholmod_di_demo.dir --rc=C:\PROGRA~2\WI3CF2~1\10\bin\100261~1.0\x64\rc.exe --mt=C:\PROGRA~2\WI3CF2~1\10\bin\100261~1.0\x64\mt.exe --manifests  -- C:\PROGRA~2\MICROS~2\2019\ENTERP~1\VC\Tools\MSVC\1429~1.301\bin\Hostx64\x64\link.exe  CMakeFiles\cholmod_di_demo.dir\Demo\cholmod_di_demo.c.obj CMakeFiles\cholmod_di_demo.dir\cmake_device_link.obj  /out:cholmod_di_demo.exe /implib:cholmod_di_demo.lib /pdb:cholmod_di_demo.pdb /version:0.0 /machine:x64 /nologo    /debug /INCREMENTAL  cholmod_static.lib  F:\PRreview\vcpkg\installed\x64-windows-static\debug\lib\suitesparseconfig_static.lib  F:\PRreview\vcpkg\installed\x64-windows-static\debug\lib\amd_static.lib  F:\PRreview\vcpkg\installed\x64-windows-static\debug\lib\colamd_static.lib  F:\PRreview\vcpkg\installed\x64-windows-static\debug\lib\camd_static.lib  F:\PRreview\vcpkg\installed\x64-windows-static\debug\lib\ccolamd_static.lib  F:\PRreview\vcpkg\installed\x64-windows-static\debug\lib\suitesparseconfig_static.lib  F:\PRreview\vcpkg\installed\x64-windows-static\debug\lib\lapack.lib  F:\PRreview\vcpkg\installed\x64-windows-static\debug\lib\blas.lib  F:\PRreview\vcpkg\installed\x64-windows-static\debug\lib\cblas.lib  F:\PRreview\vcpkg\installed\x64-windows-static\debug\lib\blas.lib  F:\PRreview\vcpkg\installed\x64-windows-static\debug\lib\cblas.lib  "C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v12.2\lib\x64\nvrtc.lib"  "C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v12.2\lib\x64\nvJitLink.lib"  "C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v12.2\lib\x64\cudart_static.lib"  "C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v12.2\lib\x64\cublas.lib"  "C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v12.2\lib\x64\cublasLt.lib"  cudadevrt.lib  cudart_static.lib  kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib -LIBPATH:"C:/Program Files/NVIDIA GPU Computing Toolkit/CUDA/v12.2/lib/x64" && cd ."
> > LINK Pass 1: command "C:\PROGRA~2\MICROS~2\2019\ENTERP~1\VC\Tools\MSVC\1429~1.301\bin\Hostx64\x64\link.exe CMakeFiles\cholmod_di_demo.dir\Demo\cholmod_di_demo.c.obj CMakeFiles\cholmod_di_demo.dir\cmake_device_link.obj /out:cholmod_di_demo.exe /implib:cholmod_di_demo.lib /pdb:cholmod_di_demo.pdb /version:0.0 /machine:x64 /nologo /debug /INCREMENTAL cholmod_static.lib F:\PRreview\vcpkg\installed\x64-windows-static\debug\lib\suitesparseconfig_static.lib F:\PRreview\vcpkg\installed\x64-windows-static\debug\lib\amd_static.lib F:\PRreview\vcpkg\installed\x64-windows-static\debug\lib\colamd_static.lib F:\PRreview\vcpkg\installed\x64-windows-static\debug\lib\camd_static.lib F:\PRreview\vcpkg\installed\x64-windows-static\debug\lib\ccolamd_static.lib F:\PRreview\vcpkg\installed\x64-windows-static\debug\lib\suitesparseconfig_static.lib F:\PRreview\vcpkg\installed\x64-windows-static\debug\lib\lapack.lib F:\PRreview\vcpkg\installed\x64-windows-static\debug\lib\blas.lib F:\PRreview\vcpkg\installed\x64-windows-static\debug\lib\cblas.lib F:\PRreview\vcpkg\installed\x64-windows-static\debug\lib\blas.lib F:\PRreview\vcpkg\installed\x64-windows-static\debug\lib\cblas.lib C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v12.2\lib\x64\nvrtc.lib C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v12.2\lib\x64\nvJitLink.lib C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v12.2\lib\x64\cudart_static.lib C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v12.2\lib\x64\cublas.lib C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v12.2\lib\x64\cublasLt.lib cudadevrt.lib cudart_static.lib kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib -LIBPATH:C:/Program Files/NVIDIA GPU Computing Toolkit/CUDA/v12.2/lib/x64 /MANIFEST /MANIFESTFILE:CMakeFiles\cholmod_di_demo.dir/intermediate.manifest CMakeFiles\cholmod_di_demo.dir/manifest.res" failed (exit code 1120) with the following output:
> >    Creating library cholmod_di_demo.lib and object cholmod_di_demo.exp
> > LINK : warning LNK4098: defaultlib 'LIBCMT' conflicts with use of other libs; use /NODEFAULTLIB:library
> > cholmod_static.lib(cholmod_super_numeric.c.obj) : error LNK2019: unresolved external symbol dtrsm__ referenced in function rd_cholmod_super_numeric_worker
> > cholmod_static.lib(cholmod_super_solve.c.obj) : error LNK2001: unresolved external symbol dtrsm__
> > cholmod_static.lib(cholmod_super_numeric.c.obj) : error LNK2019: unresolved external symbol strsm__ referenced in function rs_cholmod_super_numeric_worker
> > cholmod_static.lib(cholmod_super_solve.c.obj) : error LNK2001: unresolved external symbol strsm__
> > cholmod_static.lib(cholmod_super_numeric.c.obj) : error LNK2019: unresolved external symbol ztrsm__ referenced in function cd_cholmod_super_numeric_worker
> > cholmod_static.lib(cholmod_super_solve.c.obj) : error LNK2001: unresolved external symbol ztrsm__
> > cholmod_static.lib(cholmod_super_numeric.c.obj) : error LNK2019: unresolved external symbol ctrsm__ referenced in function cs_cholmod_super_numeric_worker
> > cholmod_static.lib(cholmod_super_solve.c.obj) : error LNK2001: unresolved external symbol ctrsm__
> > cholmod_static.lib(cholmod_super_numeric.c.obj) : error LNK2019: unresolved external symbol dgemm__ referenced in function rd_cholmod_super_numeric_worker
> > cholmod_static.lib(cholmod_super_solve.c.obj) : error LNK2001: unresolved external symbol dgemm__
> > cholmod_static.lib(cholmod_super_numeric.c.obj) : error LNK2019: unresolved external symbol sgemm__ referenced in function rs_cholmod_super_numeric_worker
> > cholmod_static.lib(cholmod_super_solve.c.obj) : error LNK2001: unresolved external symbol sgemm__
> > cholmod_static.lib(cholmod_super_numeric.c.obj) : error LNK2019: unresolved external symbol zgemm__ referenced in function cd_cholmod_super_numeric_worker
> > cholmod_static.lib(cholmod_super_solve.c.obj) : error LNK2001: unresolved external symbol zgemm__
> > cholmod_static.lib(cholmod_super_numeric.c.obj) : error LNK2019: unresolved external symbol cgemm__ referenced in function cs_cholmod_super_numeric_worker
> > cholmod_static.lib(cholmod_super_solve.c.obj) : error LNK2001: unresolved external symbol cgemm__
> > cholmod_static.lib(cholmod_super_numeric.c.obj) : error LNK2019: unresolved external symbol dsyrk__ referenced in function rd_cholmod_super_numeric_worker
> > cholmod_static.lib(cholmod_super_numeric.c.obj) : error LNK2019: unresolved external symbol ssyrk__ referenced in function rs_cholmod_super_numeric_worker
> > cholmod_static.lib(cholmod_super_numeric.c.obj) : error LNK2019: unresolved external symbol zherk__ referenced in function cd_cholmod_super_numeric_worker
> > cholmod_static.lib(cholmod_super_numeric.c.obj) : error LNK2019: unresolved external symbol cherk__ referenced in function cs_cholmod_super_numeric_worker
> > cholmod_static.lib(cholmod_super_numeric.c.obj) : error LNK2019: unresolved external symbol dpotrf__ referenced in function rd_cholmod_super_numeric_worker
> > cholmod_static.lib(cholmod_super_numeric.c.obj) : error LNK2019: unresolved external symbol spotrf__ referenced in function rs_cholmod_super_numeric_worker
> > cholmod_static.lib(cholmod_super_numeric.c.obj) : error LNK2019: unresolved external symbol zpotrf__ referenced in function cd_cholmod_super_numeric_worker
> > cholmod_static.lib(cholmod_super_numeric.c.obj) : error LNK2019: unresolved external symbol cpotrf__ referenced in function cs_cholmod_super_numeric_worker
> > cholmod_static.lib(cholmod_super_solve.c.obj) : error LNK2019: unresolved external symbol dgemv__ referenced in function rd_cholmod_super_lsolve_worker
> > cholmod_static.lib(cholmod_super_solve.c.obj) : error LNK2019: unresolved external symbol sgemv__ referenced in function rs_cholmod_super_lsolve_worker
> > cholmod_static.lib(cholmod_super_solve.c.obj) : error LNK2019: unresolved external symbol zgemv__ referenced in function cd_cholmod_super_lsolve_worker
> > cholmod_static.lib(cholmod_super_solve.c.obj) : error LNK2019: unresolved external symbol cgemv__ referenced in function cs_cholmod_super_lsolve_worker
> > cholmod_static.lib(cholmod_super_solve.c.obj) : error LNK2019: unresolved external symbol dtrsv__ referenced in function rd_cholmod_super_lsolve_worker
> > cholmod_static.lib(cholmod_super_solve.c.obj) : error LNK2019: unresolved external symbol strsv__ referenced in function rs_cholmod_super_lsolve_worker
> > cholmod_static.lib(cholmod_super_solve.c.obj) : error LNK2019: unresolved external symbol ztrsv__ referenced in function cd_cholmod_super_lsolve_worker
> > cholmod_static.lib(cholmod_super_solve.c.obj) : error LNK2019: unresolved external symbol ctrsv__ referenced in function cs_cholmod_super_lsolve_worker
> > cholmod_di_demo.exe : fatal error LNK1120: 24 unresolved externals
> > ```
> 
> I'm afraid I don't have the capacity to deal with this edge case at the moment. Would it be possible to move forward with this known failure case?

Could you please add `"supports": "!(windows & static)"` for feature `suitesparse-cholmod[cuda]`?

valgur committed:
Merge remote-tracking branch 'origin/master' into update/suitesparse

valgur committed:
[suitesparse-cholmod] disable windows-static for 'cuda'

valgur committed:
vdb

valgur committed:
Merge remote-tracking branch 'origin/master' into update/suitesparse

# Conflicts:
#	ports/g2o/vcpkg.json
#	versions/baseline.json
#	versions/g-/g2o.json

valgur committed:
vdb

valgur committed:
Merge remote-tracking branch 'origin/master' into update/suitesparse

# Conflicts:
#	ports/ceres/0003_fix_exported_ceres_config.patch
#	ports/ceres/find-package-required.patch
#	ports/ceres/portfile.cmake
#	ports/ceres/vcpkg.json
#	ports/theia/fix-external-dependencies.patch
#	ports/theia/portfile.cmake
#	ports/theia/vcpkg.json
#	versions/baseline.json
#	versions/c-/ceres.json

valgur committed:
vdb

valgur committed:
Merge remote-tracking branch 'origin/master' into update/suitesparse

# Conflicts:
#	ports/colmap/vcpkg.json
#	ports/g2o/vcpkg.json
#	scripts/ci.baseline.txt
#	versions/baseline.json
#	versions/c-/colmap.json
#	versions/g-/g2o.json

valgur committed:
vdb

valgur committed:
Merge remote-tracking branch 'origin/master' into update/suitesparse

# Conflicts:
#	ports/g2o/vcpkg.json
#	versions/baseline.json
#	versions/g-/g2o.json

valgur committed:
vdb

valgur committed:
Merge remote-tracking branch 'origin/master' into update/suitesparse

valgur commented:
@LilyWangLL @dg0yt I think everything from the review comments has been addressed and any conflicts from the recent Ceres, COLMAP and g2o PRs have been resolved as well. I would be grateful for another look.
